pkg/aggregator/aggregator.go

```diff
@@ -36,6 +37,30 @@ const bucketSize = 10                         // fixed for now
 // MetricSamplePoolBatchSize is the batch size of the metric sample pool.
 const MetricSamplePoolBatchSize = 32
 
+// tagsetSizeThresholdsCount is the size of tagsetSizeThresholds
+const tagsetSizeThresholdsCount = 2
+
+// tagsetSizeThresholds are thresholds for large tagsets.  For troubleshooting,
+// we want to know how many of these "large" tagsets we are handling for both
+// timeseries and sketches.
+var tagsetSizeThresholds = [tagsetSizeThresholdsCount]uint64{90, 100}
```

Comment:
Having a const representing "how many" thresholds we want to handle in this new logic may not be the best solution here IMO, especially since I'm not sure we'll change this number any time soon (and it would not be a big change anyway).
Defining your objects this way forces you to use this const everywhere for the slices and for the implementation 
 and it also results in duplicated ~15 lines of logic in `updateHugeSketchesTelemetry` and `updateHugeSeriesTelemetry`.

My point is that I think it is worth having a new type composed of all of these instead of all these different "independent" vars, it may be easier to read with this type also handling the logic.
These thresholds could be either provided as parameters while creating the instance (you would even be able to handle a dynamic amount of thresholds if you feel it's needed) or even hard-coded as const since they'll most likely be this way for a long time.

Not a strong opinion though.

Reply:
The advantage was that I could use arrays for the globals, rather than slices.  But you're right, it's minor.  I'll refactor into a type.